### PR TITLE
Issue 29387 remove savepublish ia actionlets

### LINKED ISSUE #29387: Remove save and publish functionality from the AI Actionlets
### Parent Issue

https://github.com/dotCMS/core/issues/29380

### User Story

As an user I want to be able to decide which is the next sub action after the IA generation

### Acceptance Criteria

Currently the IA WF actionlets does save + publish, the idea would be to leave the ability to the user to decide if wants or not save + publish or something else.

### Proposed Objective

Core Features

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by jdotcms**
Removing save actions from IA actionlets

This PR fixes: #29387

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/tag/business/TagFactoryImpl.java [hunk 1]
```diff
@@ -651,6 +654,40 @@ public List<Tag> getTagsByInodeAndFieldVarName(String inode, String fieldVarName
...
+                        "UNION ALL " +
+                        "   select lower(tagname) as tagname, tag_inode.inode as tinode " +
+                        "   from  " +
+                        "   tag, tag_inode  " +
+                        "   where  " +
+                        "   tag.tag_id=tag_inode.tag_id  " +
+                        ") as foo  " +
+                        "group by tagname  " +
+                        "order by count(tinode) desc " +
+                        "limit 1000";
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/tag/business/TagFactoryImpl.java:680]: Shouldn't this limit be customizable via a variable passed down in the method's signature?

**Review by jcastro-dotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/ai/api/DotAIAPI.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.dotcms.ai.api;
+
+/**
+ * Provides a facade for all artificial intelligence services offered by dotCMS.
+ * @author jsanca
+ */
+public interface DotAIAPI {
+
+    CompletionsAPI getCompletionsAPI(Object... initArguments);
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/ai/api/DotAIAPI.java:9]: Isn't this the same as the previous at `CompletionsAPIProvider` class?

File: dotcms-integration/src/test/java/com/dotcms/ai/workflow/OpenAIAutoTagActionletTest.java [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+        final Map<String, WorkflowActionClassParameter> params = Map.of(
+                OpenAIParams.OVERWRITE_FIELDS.key, new WorkflowActionClassParameter("true"),
+                OpenAIParams.LIMIT_TAGS_TO_HOST.key, new WorkflowActionClassParameter("false"),
+                OpenAIParams.MODEL.key, new WorkflowActionClassParameter("gpt-3.5-turbo"),
+                OpenAIParams.TEMPERATURE.key, new WorkflowActionClassParameter("0.5")
+        );
+
+        new OpenAIAutoTagActionlet().executeAction(processor, params);
+
+        final Object tagsObject = contentlet.get("tags");
```
Comment (dsilvam) [inline dotcms-integration/src/test/java/com/dotcms/ai/workflow/OpenAIAutoTagActionletTest.java:250]: Nice test!

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/ai/util/ContentToStringUtil.java [hunk 4]
```diff
@@ -236,7 +236,7 @@ private Optional<String> parseFile(@NotNull File file) {
      *
      * @return An {@link Optional} with the appended Strings from each field value.
      */
-    public Optional<String> parseFields(@NotNull final Contentlet contentlet, @NotNull final List<Field> fields) {
+    public Optional<String>     parseFields(@NotNull final Contentlet contentlet, @NotNull final List<Field> fields) {
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/ai/util/ContentToStringUtil.java:239]: Spaces

File: dotCMS/src/main/java/com/dotcms/ai/api/DotAIAPI.java (hunk 2 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/ai/api/DotAIAPI.java:9]: nvm, I got it

**Review by victoralfaro-dotcms — APPROVED** [review]

**Pull request merged**